### PR TITLE
feat(napi/transform): add `TransformOptions::target` API

### PR DESCRIPTION
**Pull request by Boshen**

### CODE REVIEW

**Comment by Boshen** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7426?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#7426** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7426?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7426?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#7424** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7424?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

File: napi/transform/test/transform.test.ts [hunk 1]
```diff
@@ -50,6 +50,33 @@ describe('transform', () => {
...
+
+  test.each(data)('transform %s', (target, code) => {
+    // Also test array syntax.
+    const ret = oxc.transform('test.js', code, { target: [target] });
+    assert(ret.errors.length == 0);
+    assert(ret.code);
+    assert.notEqual(ret.code, code);
+  });
+
+  test.each(data)('no transform esnext', (_target, code) => {
```
Comment (overlookmotel) [inline napi/transform/test/transform.test.ts:72]: Nit:

```suggestion
  test.each(data)('no transform esnext: %s', (_target, code) => {
```

File: napi/transform/index.d.ts [hunk 2]
```diff
@@ -203,6 +203,21 @@ export interface TransformOptions {
...
+   * Example:
+   *
+   * * 'es2015'
+   * * ['es2020', 'chrome58', 'edge16', 'firefox57', 'node12', 'safari11']
+   *
+   * @default `esnext` (No transformation)
+   *
+   * @see [esbuild#target](<https://esbuild.github.io/api/#target)
+   */
+  target?: string | Array<string>
```
Comment (Dunqing) [inline napi/transform/index.d.ts:220]: Can we make it type-safe? For example: 
```ts
{
 target: 'es2015' | 'es2016' | Array<`chrome${string}` | `edge${string}`>
}
```

Comment (Boshen) [reply napi/transform/index.d.ts:220]: It would be

```
type Target = 'es2015' | ... 
Target | Array<Target>
```

I'll do this later when I get time.

**Pull request merged**